posthog/api/person.py

```diff
@@ -209,7 +209,7 @@ def retention(self, request: request.Request) -> response.Response:
                 {"message": "Could not retrieve team", "detail": "Could not validate team associated with user"},
                 status=400,
             )
-        filter = RetentionFilter(request=request)
+        filter = RetentionFilter(request=request, team=team)
```

Comment:
non-blocking

We could make this non-optional, that way we'd catch any other places we're missing the arg. Unless of course it's not required. Also, I guess team is actually derived from request, this we could enforce the desired invariant by `RetentionFilter` using the request derived team directly.

Reply:
Yep, plan was to make this required eventually :)